tests/test_keras_integration.py

```diff
@@ -178,12 +179,56 @@ def test_save_pretrained(self):
 
         model.build((None, 2))
 
-        save_pretrained_keras(model, f"{WORKING_REPO_DIR}/{REPO_NAME}")
+        save_pretrained_keras(
+            model,
+            f"{WORKING_REPO_DIR}/{REPO_NAME}",
+        )
         files = os.listdir(f"{WORKING_REPO_DIR}/{REPO_NAME}")
 
         self.assertIn("saved_model.pb", files)
         self.assertIn("keras_metadata.pb", files)
         self.assertEqual(len(files), 4)
+        model = from_pretrained_keras(f"{WORKING_REPO_DIR}/{REPO_NAME}")
+        self.assertIsNone(model.optimizer)
```

Comment:
I would suggest to avoid testing this as well in the same test. Maybe let's have a test which is
`test_save_pretrained_does_not_save_optimizer_state_by_default` which has this